src/mesh/unstructuredMesh/BoundaryMesh.cpp

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+
+#include "NeoFOAM/mesh/unstructuredMesh/BoundaryMesh.hpp"
+
+namespace NeoFOAM
+{
+
+BoundaryMesh::BoundaryMesh(
+    const executor& exec,
+    labelField faceCells,
+    vectorField Cf,
+    vectorField Cn,
+    vectorField Sf,
+    scalarField magSf,
+    vectorField nf,
+    vectorField delta,
+    scalarField weights,
+    scalarField deltaCoeffs,
+    std::vector<localIdx> offset
+)
+    : exec_(exec),
+      faceCells_(faceCells),
+      Cf_(Cf),
+      Cn_(Cn),
+      Sf_(Sf),
+      magSf_(magSf),
+      nf_(nf),
+      delta_(delta),
+      weights_(weights),
+      deltaCoeffs_(deltaCoeffs),
+      offset_(offset) {
+
+      };
+
+// Accessor methods
+const labelField& BoundaryMesh::faceCells() const
+{
+    return faceCells_;
+}
+
+std::span<const label> BoundaryMesh::faceCells(const int i) const
```

Comment:
IMO doesn't need to be const if we don't pass in references.

```suggestion
std::span<const label> BoundaryMesh::faceCells(label i) const
```